src/components/TrackTile/hooks/useAxiosTrackTileService.ts

```diff
@@ -92,6 +92,94 @@ export const useAxiosTrackTileService = (): TrackTileService => {
     return cache.trackerValues[valuesContextKey];
   };
 
+  const hasCodeInOntology = (
+    relationship: CodedRelationship,
+    code: string,
+  ): boolean => {
+    return relationship.specializedBy.some(
+      (r) => r.code === code || hasCodeInOntology(r, code),
+    );
+  };
+
+  const extractMetricCoding = (
+    codings: Observation['node']['code']['coding'],
+  ) => {
+    // If the record has a tracker system, short-circuit this lookup.
+    const trackerCoding = codings.find(
+      ({ system }) =>
+        system === TRACKER_CODE_SYSTEM || system === TRACKER_PILLAR_CODE_SYSTEM,
+    );
+    if (trackerCoding) {
+      return trackerCoding;
+    }
+
+    const relationshipGroups = Object.values(cache.ontologies);
+    for (const coding of codings) {
+      for (const relationships of relationshipGroups) {
+        const trackerMatch = relationships.find((r) => {
+          return hasCodeInOntology(r, coding.code);
+        });
+        if (trackerMatch) {
+          return trackerMatch;
+        }
+      }
+    }
```

Comment:
This diff from lines 95 - 182 looks much bigger than it is.  It's really just moving the functions `extractMetricCoding`, `extractMetricId`, and `extractTrackerValues` to be inside the hook.  The real diff is here in lines 116 - 126, which utilizes `hasCodeInOntology` and the utilization of `cache.ontologies` (which is the primary reason for moving these into the hook).